pontoon/administration/tests/test_sync_projects.py

```diff
@@ -0,0 +1,655 @@
+        assert_not_in(self.main_db_translation, self.changeset.translations_to_update)
+
+    def test_update_db_approve_translation(self):
+        """
+        Approve any un-approved translations that have counterparts in
+        VCS.
+        """
+        self.create_db_entities_translations()
+        self.main_db_translation.approved = False
+        self.main_db_translation.approved_date = None
+        self.main_db_translation.save()
+
+        self.update_main_db_entity()
+        self.main_db_translation.refresh_from_db()
+        assert_attributes_equal(
+            self.main_db_translation,
+            approved=True,
+            approved_date=datetime(1970, 1, 1)
+        )
+
+    def test_update_db_new_translation(self):
+        """
+        If a matching translation does not exist in the database, create a new
+        one.
+        """
+        self.create_db_entities_translations()
+        self.main_db_translation.delete()
+        self.update_main_db_entity()
+
+        translation = self.main_db_entity.translation_set.all()[0]
+        assert_attributes_equal(
+            translation,
+            locale=self.translated_locale,
+            string='Translated String',
+            plural_form=None,
+            approved=True,
+            approved_date=datetime(1970, 1, 1),
+            fuzzy=False,
+            extra={'tags': []}
+        )
+
+    def test_update_db_unapprove_existing(self):
+        """
+        Any existing translations that don't match anything in VCS get
+        unapproved.
+        """
+        self.create_db_entities_translations()
+
+        self.main_db_translation.approved = True
+        self.main_db_translation.approved_date = datetime(1970, 1, 1)
+        self.main_db_translation.approved_user = UserFactory.create()
+        self.main_db_translation.save()
+        self.main_vcs_translation.strings[None] = 'New Translated String'
+
+        self.update_main_db_entity()
+        self.main_db_translation.refresh_from_db()
+        assert_attributes_equal(
+            self.main_db_translation,
+            approved=False,
+            approved_user=None,
+            approved_date=None
+        )
+
+    def test_update_db_unapprove_clean(self):
+        """
+        If translations that are set to be unapproved were already unapproved,
+        don't bother updating them.
+        """
+        self.create_db_entities_translations()
+
+        self.main_db_translation.approved = False
+        self.main_db_translation.approved_date = None
+        self.main_db_translation.approved_user = None
+        self.main_db_translation.save()
+        self.main_vcs_translation.strings[None] = 'New Translated String'
+
+        self.update_main_db_entity()
+        self.main_db_translation.refresh_from_db()
+        assert_not_in(self.main_db_translation, self.changeset.translations_to_update)
+
+    def test_obsolete_db(self):
```

Comment:
I can't think of a docstring that provided any more info than the name of the test itself.